pkg/controller/nodelifecycle/node_lifecycle_controller_test.go

```diff
@@ -2847,6 +2851,12 @@ func TestSwapUnreachableNotReadyTaints(t *testing.T) {
 	if err := nodeController.monitorNodeHealth(ctx); err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
+
+	// Remove the NoExecute taint rate limiter. If we don't do this, not all the
+	// nodes will be processed and tainted appropriately when doing the NoExecute taint pass.
+	for k := range nodeController.zoneNoExecuteTainter {
```

Comment:
This code snippet duplicates multiple times, should we extract it and make it become a helper function?